website/docs/learn-graphql/guides/thinking-in-graphql.md

```diff
@@ -0,0 +1,147 @@
+---
```

Comment:
I would argue that this page should actually come before the "Design of GraphQL" page. Its concepts are broader and don't depend on the fragment composition which is the focus of that page. But especially, this would resolve my concern on the other page that there is an unexplained assumption that you wouldn't be issuing queries from multiple levels of the component tree. This page explains why it's important for all of the data needs to roll up into one "broad query". If this page comes first, it will make more sense to work through how to keep the components loosely coupled even when making one broad query in the other page.

Reply:
Oh, interesting thought. I see what you're saying. I guess the chronological order just follows so naturally to me, given the context I already have, but it would not be the first time that I have to learn that with different experiences (eg in TMP) what feels natural is the opposite.

Will try it out!

Reply:
@butaud Ok, pushed. I'll let it bake for a bit; let me know what you think.